pkg/crypto/walletapi/wallet_api.go

```diff
@@ -56,6 +65,33 @@ type Crypto interface {
 	//
 	//		in case of error, the index will be -1
 	FindVerKey(candidateKeys []string) (int, error)
+
+	// ConvertToEncryptionKey will create and persist a Curve25519 encryption keypair
+	// created from the ed25519 keypair with the provided public key.
+	//
+	// Returns:
+	//
+	// []byte: The public key of the encryption keypair created. nil in case of error.
+	//
+	// error:
+	ConvertToEncryptionKey(pubkey []byte) ([]byte, error)
+
+	// AttachCryptoOperator attaches a crypto operator to this wallet, so the operator can use its private keys.
+	AttachCryptoOperator(cryptoOp KeyHolderUser) error
+}
+
+// KeyHolderUser denotes an object that accesses keys in a KeyHolder
+// A KeyHolderUser must be defined inside pkg/crypto to fulfill the
+// interface, which ensures that access to private keys is limited
+// to only packages in pkg/crypto
+type KeyHolderUser interface {
+
+	// InjectKeyHolder injects a KeyHolder which will provide private keys for the KeyHolderUser's crypto computations.
+	// Usage note: InjectKeyHolder is not for the client to call,
```

Comment:
this is a good example of comment.. to be copied for the above functions I mentioned

Reply:
Talked to @troyronda and he pointed out that we need to allow clients to define their own CryptoOperators to plug in the crypto suites they need to use, so KeyHolder is now part of the public API, and this is now implementable by client code.